runtime/minor_gc.c

```diff
@@ -721,6 +707,23 @@ caml_stw_empty_minor_heap_no_major_slice(caml_domain_state* domain,
     CAML_EV_END(EV_MINOR_LEAVE_BARRIER);
   }
 
+  /* Finalize dead custom blocks; do the accounting for the live ones.
+     This must be done after all domains have finished minor GC, but
+     before this domain resumes running OCaml code. */
+  struct caml_custom_elt *elt;
+  for (elt = domain->minor_tables->custom.base;
+       elt < domain->minor_tables->custom.ptr; elt++) {
+    value *v = &elt->block;
+    if (Is_block(*v) && Is_young(*v)) {
+      if (get_header_val(*v) == 0) { /* value copied to major heap */
+        caml_adjust_gc_speed(elt->mem, elt->max);
+      } else {
+        void (*final_fun)(value) = Custom_ops_val(*v)->finalize;
+        if (final_fun != NULL) final_fun(*v);
+      }
+    }
+  }
+
   CAML_EV_BEGIN(EV_MINOR_FINALIZERS_ADMIN);
```

Comment:
The code above must come under this span for proper accounting.

Reply:
I understand the suggestion as moving the `custom_finalize_minor` call above, with or without its `EV_MINOR_FINALIZED` event bracketing, inside the `EV_MINORS_FINALIZERS_ADMIN` event bracketing. I discussed this with @damiendoligez and we concluded that the PR as-is is correct:

1. `EV_MINOR_FINALIZED` is new in this PR, it was not used in the runtime before the change. It is part of those runtime events who were basically dropped in the 4.x -> 5.x transition, and @damiendoligez believes that they were used for something similar to what he is introducing now, so the reuse is appropriate. 
2. `EV_MINOR_FINALIZERS_ADMIN` is about administrative book-keeping of data structures related to *OCaml* finalized values, while `EV_MINOR_FINALIZED` above is about the finalization of *C* finalizers (custom blocks with a finalization operation), so they better be kept separate. (The names are not great, it is easy to get confused, but reusing the existing names is a reasonable choice.)